IntegrationTests/tests_01_allocation_counters/test_01_resources/test_1k_requests.swift

```diff
@@ -0,0 +1,215 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHPACK
+import NIOHTTP2
+
+struct ServerOnly1KRequestsBenchmark {
+    private let concurrentStreams: Int
+
+    private let channel: EmbeddedChannel
+
+    // This offset is preserved across runs.
+    var streamID = HTTP2StreamID(1)
+
+    // A manually constructed data frame.
+    private var dataFrame: ByteBuffer? = {
```

Comment:
Why are `dataFrame` and `headersFrame` optional?

Reply:
Oh, that's a remnant from an older design where I thought I was going to swap them with nil. Will fix.